### PR TITLE
enhance(notes): change dendron id format to be alphanumeric lowercase

### PR DESCRIPTION
**Pull request by kevinslin**
This brings more compatibility to the widest possible use cases when synchronizing dendron notes with other sources.

For example, when ids are used in urls, certain hosting providers like netlify will lowercase all urls which will break users that publish using case sensitive ids.

This update makes the following changes:

- update dendron ids to be alphanumeric lowercase
- removes nanoidDictionary dependency
- extend the default length of a dendron id to 32 characters (this is the same length as what other providers like notion and loom use )

Documentation available at https://github.com/dendronhq/dendron-site/pull/402
***
# Dendron Extended PR Checklist

### Basics

- [x] code should follow [Code Conventions](dev.process.code)
- [~] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](dev.process.code.best-practices).
- [x] sticking to existing conventions instead of creating new ones [^eg-existing-convention]

### Extended

- General
  - [x] check whether code be simplified
  - [x] check if similar function already exist in the codebase. if so, can it be re-used?
  - [x] check if this change adversely impact performance
- Operations
  - [x] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [x] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed [^eg-check-for-design-review]

## Instrumentation

NA

### Basics

- [~] [Write Tests](dev.process.qa.test)  -> we don't have setup to exhaustively test uuid is lowercase
- [x] [Confirm existing tests pass](dev.process.qa.test)
- [x] [Confirm manual testing](dev.process.qa.test) 
- [~] Common cases tested
- [~] 1-2 Edge cases tested
- [~] If your tests changes an existing snapshot, snapshots have been [updated](dev.process.qa.test)

### Extended

NA

### Basics

- [x] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR

## 

### Basics

- [~] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](dev) or [Packages](pkg)

## 

### Extended

- [~  is this a developer BREAKING change? if another person cloning from this branch will need to adjust their dependencies or mental model of the architecture, then it is. if this is the case, make sure this is communicated according to [Close Loop](dev.process.close-loop)

### CODE REVIEW

**Comment by jonathanyeung** [comment]
This code looks fine - were we still planning on making the engine interpret id's in a case-insensitive manner?

**Review by jonathanyeung — APPROVED** [review]

**Comment by kevinslin** [comment]
> This code looks fine - were we still planning on making the engine interpret id's in a case-insensitive manner?

So I took a look at the work involved and its actually going to be a moderate lift (eg. a full day). to move there iteratively, I'm thinking we should make this the default and then introduce doctor command to format existing ids (ids can already be incompatible due to `-` symbols which were accepted before)

File: packages/common-all/src/uuid.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-import { alphanumeric } from "nanoid-dictionary";
 
 /** Using this length, according to [nanoid collision calculator](https://zelark.github.io/nano-id-cc/),
  * generating 1000 IDs per hour, it would take around 919 years to have 1 percent chance of a single collision.
  * This is okay for the "insecure" generator, which is used in limited cases where collisions are less likely.
  */
 const SHORT_ID_LENGTH = 12;
 /** Default length for nanoids. */
-const LONG_ID_LENGTH = 21;
+const LONG_ID_LENGTH = 32;
```
Comment (SeriousBug) [inline packages/common-all/src/uuid.ts:10]: Using the collision calculator linked in the comment, bumping the ID length from 21 to 23 with the dictionary change in this PR is enough to preserve the collision resistence. Shorter IDs mean shorter links, so I'd suggesst lowering this to 23.

**Review by SeriousBug — CHANGES_REQUESTED** [review]
Looks good, but I'd suggest a shorter ID length. We don't have to go all the way to 32 to preserve the collision resistence.

File: packages/common-all/src/uuid.ts (hunk 1 above)
Comment (kevinslin) [reply packages/common-all/src/uuid.ts]: updated!

**Pull request merged**